### ISSUE TITLE
bug: Query data cannot be undefined error on index page

### ISSUE THREAD

**Issue by kwerenta** [t3-oss/create-t3-app#1943]
### Provide environment information

System:
    OS: macOS 15.0
    CPU: (8) arm64 Apple M1
    Memory: 455.50 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
Binaries:
    Node: 18.18.2 - ~/.nvm/versions/node/v18.18.2/bin/node
    Yarn: 1.22.15 - ~/.yarn/bin/yarn
    npm: 9.8.1 - ~/.nvm/versions/node/v18.18.2/bin/npm
    bun: 1.1.15 - ~/.bun/bin/bun
    
   ct3aMetdata.initVersion - 7.36.1

### Describe the bug

When you create new project with options like below and you log in using discord, an error will appear. 
In browser:
**Unhandled Runtime Error.** `Error: [["post","getLatest"],{"type":"query"}] data is undefined`
In console:
```
Query data cannot be undefined. Please make sure to return a value other than undefined from your query function. Affected query key: [["post","getLatest"],{"type":"query"}]
 ⨯ Internal error: Error: [["post","getLatest"],{"type":"query"}] data is undefined
    at Object.onSuccess (./node_modules/@tanstack/query-core/build/modern/query.js:241:19)
    at resolve (./node_modules/@tanstack/query-core/build/modern/retryer.js:58:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
 ```

### Reproduction repo

Newly created project

### To reproduce

1. Create project using `bun create t3-app@latest`. Select next-auth, drizzle, tailwind, trpc, next.js app router and sqlite as database.
2. Run `bun run db:push`
3. Start server using `bun dev`
4. Log in to your discord account
5. Observe the error: **Unhandled Runtime Error.** `Error: [["post","getLatest"],{"type":"query"}] data is undefined`

### Additional information

After replacing `const [latestPost] = api.post.getLatest.useSuspenseQuery();` in `src/app/_components/post.tsx` with `const { data: latestPost } = api.post.getLatest.useQuery();` and commenting out `void api.post.getLatest.prefetch();` in `src/app/page.tsx`, it works as expected.

Another solution is that you just manually add post using Drizzle Studio, but I don't think this is expected behaviour.

**Comment by IcTxDiogo** [comment]
same  error here, i fix then add on, 

getLatest: publicProcedure.query(async ({ ctx }) => {
    const post = await ctx.db.query.posts.findFirst({
      orderBy: (posts, { desc }) => [desc(posts.createdAt)],
    });
    if (post === undefined) {
      return null;
    }
    return post;
  }),
  
  checking all returns that can be undefined to null

**Comment by datasalaryman** [comment]
> same error here, i fix then add on,
> 
> getLatest: publicProcedure.query(async ({ ctx }) => { const post = await ctx.db.query.posts.findFirst({ orderBy: (posts, { desc }) => [desc(posts.createdAt)], }); if (post === undefined) { return null; } return post; }),
> 
> checking all returns that can be undefined to null

I'll make a PR for this if no one wants

**Comment by exectx** [comment]
The merged solution (#1945 ) doesn't resolve the promise before checking for undefined
